### PR TITLE
fix: Iceberg schema evolution fails for map, array and struct types

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Bugfix

### Detail
- Iceberg schema evolution fails for map, array, and struct types

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by malachi-constant — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e7693913a3463baef745db09054e242a43565b45
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=79063833-4990-4705-b1ae-462e7a88a755%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e7693913a3463baef745db09054e242a43565b45
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3be138d7-ad22-41a4-ba2c-013807699546%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 7a8183a072569a549e4f9681660a71b570d8db82
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=109ae14f-1877-41d3-ba3f-868c85467e9e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tests/unit/test_athena_iceberg.py [hunk 1]
```diff
@@ -924,3 +924,56 @@ def test_to_iceberg_uppercase_columns(
...
+    path: str,
+    path2: str,
+    glue_database: str,
+    glue_table: str,
+) -> None:
+    df_with_col = pd.DataFrame(
+        {
+            "partition": [1, 1, 2, 2],
+            "column2": ["A", "B", "C", "D"],
+            "map_col": [{"s": "d"}, {"s": "h"}, {"i": "l"}, {}],
```
Comment (aockel) [inline tests/unit/test_athena_iceberg.py:939]: Hi, could we test this for struct fields as well? I have similar issues with struct<a: string, struct<b: string>>, failing with the exact same error.

Comment (aockel) [reply tests/unit/test_athena_iceberg.py:939]: A example column could be 
`
'struct_col': [{'attr1': 'val1', 'attr2': {'attr21': 'val21'}},
                       {'attr1': 'val1', 'attr2': {'attr21': None}},
                       {'attr1': 'val1', 'attr2': None},
                       {},
                       ]
`

File: awswrangler/_data_types.py [hunk 2]
```diff
@@ -375,7 +375,7 @@ def athena2pandas(dtype: str, dtype_backend: str | None = None) -> str:  # noqa:
         return "decimal" if dtype_backend != "pyarrow" else "double[pyarrow]"
     if dtype in ("binary", "varbinary"):
         return "bytes" if dtype_backend != "pyarrow" else "binary[pyarrow]"
-    if dtype in ("array", "row", "map"):
+    if any(dtype.startswith(t) for t in ["array", "row", "map"]):
```
Comment (aockel) [inline awswrangler/_data_types.py:378]: Open question: Do I see that right, that 'struct' type is not handled at all at the moment?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 83c3a129f954ce631e1a3a4d8f9747d41053225a
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e963ad71-a986-4d31-bc1e-d2659278b42a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Thanks!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 83c3a129f954ce631e1a3a4d8f9747d41053225a
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f3efc4c2-9669-41ea-aa61-735bc9d97512%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**